pkg/controller/migplan/migplan_controller.go

```diff
@@ -389,43 +391,46 @@ func (r *ReconcileMigPlan) planSuspended(plan *migapi.MigPlan) error {
 
 func (r *ReconcileMigPlan) AddRunningLabelForPlan(plan *migapi.MigPlan) error {
 	migrations, err := plan.ListMigrations(r)
+	runningMigrations := 0
+	needsUpdate := false
 	if err != nil {
 		return liberr.Wrap(err)
 	}
 
-	if len(migrations) == 0 {
+	//Check if there are any associated migrations that are running
+	for _, m := range migrations {
+		if m.Status.HasCondition(migctl.Running) {
+			runningMigrations++
+		}
+	}
+
+	//No migrations associated with the migplan or there are no running migrations for the Migplan
+	if len(migrations) == 0 || runningMigrations == 0 {
+		if plan.Labels != nil {
+			if _, found := plan.Labels[MigplanRunning]; found {
+				delete(plan.Labels, MigplanRunning)
+				needsUpdate = true
+			}
+		}
+		return nil
+	}
+
+	//Migplan is associated with Running migrations
+	if runningMigrations > 0 {
```

Comment:
if is not neded here the above if will always return if `runningMigrations == 0`

Reply:
well you need it nevermind, you could use a switch case here, 

```
switch {
    case runningMigrations == 0:
          // set your variables
    case runingMigrations > 0:
          // set vars
}
```